modules/nixos/common/default.nix

```diff
@@ -29,7 +29,7 @@
 
   srvos.flake = inputs.self;
 
-  boot.kernelPackages = pkgs.linuxPackages_latest;
+  boot.kernelPackages = pkgs.lib.mkIf (!config.boot.supportedFilesystems.zfs or false) pkgs.linuxPackages_latest;
```

Comment:
```suggestion
  boot.kernelPackages = if config.boot.supportedFilesystems.zfs or false then pkgs.zfsUnstable.latestCompatibleLinuxPackages else pkgs.linuxPackages_latest;
```

Reply:
Following the stable kernel is intentional, don't want kernel versions downgrades, e.g. in our next nixpkgs bump it'll go from 6.7 -> 6.6.